pkg/p2p/server_stream_handle.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package p2p
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pingcap/errors"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/proto/p2p"
+	"go.uber.org/atomic"
+)
+
+// streamHandle is used to provide the server a non-blocking handle of a stream.
+// It is needed because grpc-go does not provide a stream easily cancellable.
+// Since Go is not friendly with the receiver closing a channel, we need this
+// auxiliary data structure to make the stream cancellation more graceful.
+type streamHandle struct {
+	mu     sync.RWMutex
```

Comment:
Better to add comments about which variables are being protected by `mu`, I find the protected variables are not same in `Send` and `Close`.

Reply:
Technically it protects only `sendCh`.